### ISSUE TITLE
6.0.1 update breaks route-restful option

### ISSUE THREAD

**Issue by gdibble** [hapijs/crumb#80]
An `||` got change to an `&&`:

![image](https://cloud.githubusercontent.com/assets/906846/15259754/feff756a-1907-11e6-9329-a6e36fa8f980.png)

**Comment by stongo** [comment]
isn't the `&&` correct? the route setting should precede the plugin setting in the check and both conditions should pass.

**Comment by gdibble** [comment]
I thought so too, but in application my routes which specify...
```
    config: {
        plugins: {
            crumb: {
                restful: false
            }
        }
    }
```

...stopped working (whereas these send a form-field-value instead of req. header) when I upgraded to v6.0.1

Sorry if I've misintrepreted anything, I just couldn't get it to work w/o changing that `&&` => `||`

**Comment by aef-** [comment]
We need to add a condition that explicitly checks for `false` as the value and to short-circuit the plugin/settings config condition.

I think this should do the trick (test ln 659 should expect 200 instead of 403, this causes it to fail appropriately)
```
const routeIsRestful = request.route.settings.plugins._crumb ? 
                       request.route.settings.plugins._crumb.restful : undefined;
 if (routeIsRestful === false ||
     (routeIsRestful !== true && settings.restful === false))
```

**Comment by stongo** [comment]
The pull request of branch `route-config` should fix this. Can anyone confirm?

**Comment by stongo** [comment]
Fixed by release 6.0.2

**Comment by gdibble** [comment]
Many thanks
